### PR TITLE
Remove triage/waiting-for-ci after PR is merged

### PR DESCRIPTION
**Pull request by saumya1singh**
Hi @FroMage @gsmet :) 

This is the PR for #29

### CODE REVIEW

File: src/main/java/io/quarkus/bot/MarkClosedPullRequestInvalid.java [hunk 1]
```diff
@@ -23,6 +25,18 @@ void markClosedPullRequestInvalid(@PullRequest.Closed GHEventPayload.PullRequest
         GHPullRequest pullRequest = pullRequestPayload.getPullRequest();
 
         if (pullRequest.isMerged()) {
+            if (!pullRequest.getLabels().isEmpty()) {
```
Comment (gsmet) [inline src/main/java/io/quarkus/bot/MarkClosedPullRequestInvalid.java]: You can drop that test given the other advices below (as the log will be inside the logic, you won't need this anymore).

File: src/main/java/io/quarkus/bot/MarkClosedPullRequestInvalid.java [hunk 2]
```diff
@@ -23,6 +25,18 @@ void markClosedPullRequestInvalid(@PullRequest.Closed GHEventPayload.PullRequest
...
+                if (!quarkusBotConfig.isDryRun()) {
```
Comment (gsmet) [inline src/main/java/io/quarkus/bot/MarkClosedPullRequestInvalid.java]: This test should be around line 33 and you should log the action only if it does something so the LOG.info() line will be inside the if of line 32.

File: src/main/java/io/quarkus/bot/MarkClosedPullRequestInvalid.java [hunk 3]
```diff
@@ -23,6 +25,18 @@ void markClosedPullRequestInvalid(@PullRequest.Closed GHEventPayload.PullRequest
...
+                    Collection<GHLabel> labels = pullRequest.getLabels();
+                    for (GHLabel label : labels) {
+                        if (label.getName().equals(Labels.TRIAGE_WAITING_FOR_CI)) {
+                            pullRequest.removeLabels(Labels.TRIAGE_WAITING_FOR_CI);
```
Comment (gsmet) [inline src/main/java/io/quarkus/bot/MarkClosedPullRequestInvalid.java]: You can `break;` from the for loop as soon as you removed it as you won't find a second one (labels are unique).

**Review by gsmet — CHANGES_REQUESTED** [review]
Hey Saumya,

The logic looks good. I added some comments to improve it.

One thing though: we try to keep each rule self contained so that we can easily tweak/drop one without breaking another.

So the idea would be to create your own `RemoveWaitingForCiLabelWhenMerged` class.

Once this is done, next step would be to add a test. The test framework is a bit tricky so maybe let's discuss it together once the first iteration is in.

**Comment by gsmet** [comment]
Better check the comments on the Files changed page, they will make more sense with all the logic visible: https://github.com/quarkusio/quarkus-bot/pull/46/files .

**Comment by saumya1singh** [comment]
> One thing though: we try to keep each rule self contained so that we can easily tweak/drop one without breaking another.
> So the idea would be to create your own `RemoveWaitingForCiLabelWhenMerged` class.
> 
Yes, I created a separate class `RemoveWaitingForCiLabelWhenMerged`, and added the logic of this rule there. 

> Once this is done, next step would be to add a test. The test framework is a bit tricky so maybe let's discuss it together once the first iteration is in.

okay, sure :)

**Review by gsmet — APPROVED** [review]
I made a small adjustment because in the end I think we can remove the label for every closed PR (even if not merged). It might happen that the label is added and in the end the PR doesn't pass CI and we close it as invalid. My fault, I hadn't thought of that when I created the issue. I just removed a if so that's no biggie.

I also added a formatter config to the project. So now, when you do mvn clean install, it will format the code with our conventions (typically 4 spaces for indentation instead of 3).

**Comment by gsmet** [comment]
And merged, thanks!

**Pull request merged**